pandas/core/internals/construction.py

```diff
@@ -595,22 +602,82 @@ def _list_of_dict_to_arrays(data, columns, coerce_float=False, dtype=None):
     data = [(type(d) is dict) and d or dict(d) for d in data]
 
     content = list(lib.dicts_to_array(data, list(columns)).T)
-    return _convert_object_array(
-        content, columns, dtype=dtype, coerce_float=coerce_float
-    )
+    columns = _validate_or_indexify_columns(content, columns)
+    content = _convert_object_array(content, dtype=dtype, coerce_float=coerce_float)
+    return content, columns
 
 
-def _convert_object_array(content, columns, coerce_float=False, dtype=None):
+def _validate_or_indexify_columns(
+    content: List, columns: Union[Index, List, None]
+) -> Union[Index, List[Axis]]:
+    """If columns is None, make numbers as column names; If not None, validate if
+    columns are valid in length.
+
+    Parameters
+    ----------
+    content: list of processed data records
```

Comment:
what does "processed data records" mean here?

Reply:
sorry for misleading, just because this is not original input data (input data being transformed a bit), so i put that word. I will just use `list of data` instead to avoid confusion.